AlphaWallet/ScrollableSegmentedControl/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,270 @@
+//
+//  ScrollableSegmentedControl.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 27/12/21.
+//
+
+import UIKit
+
+struct ScrollableSegmentedControlConfiguration {
+    let lineConfiguration: ScrollableSegmentedControlHighlightableLineViewConfiguration
+    let isProportionalWidth: Bool
+    let cellSpacing: CGFloat
+    let alignmentWhenNotScrollable: ScrollableSegmentedControl.Alignment
+    let animationDuration: TimeInterval
+    let animationCurve: UIView.AnimationCurve
+}
+
+class ScrollableSegmentedControl: UIControl {
+    enum Selection {
+        case unselected
+        case selected(Int)
+    }
+    // Only if not scrollable (total cells width + spacing < control width).
+    enum Alignment {
+        case leading
+        case trailing
+        case centered
+        case filled
+    }
+    // MARK: - Properties
+    // MARK: Private
+    private let cells: [ScrollableSegmentedControlCell]
+    private let configuration: ScrollableSegmentedControlConfiguration
+    private var intrinsicContentWidth: CGFloat = 0.0
+    private var previousHeight: CGFloat = 0.0
+    private var previousWidth: CGFloat = 0.0
+    private var scrollViewHeightConstraint: NSLayoutConstraint?
+    private var scrollViewPositionConstraints: [NSLayoutConstraint] = []
+    private var _selectedSegment: Selection = .unselected
+    // MARK: Public
+    var selectedSegment: Selection {
+        return _selectedSegment
+    }
+    // MARK: - UI Elements
+    private lazy var scrollView: UIScrollView = {
+        let scrollView = UIScrollView()
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        return scrollView
+    }()
+    private lazy var line: ScrollableSegmentedControlHighlightableLineView = {
+        let line = ScrollableSegmentedControlHighlightableLineView(configuration: self.configuration.lineConfiguration)
+        line.translatesAutoresizingMaskIntoConstraints = false
+        return line
+    }()
+    // MARK: - Constructors
+    // Proportional means variable width of cells due to content, if not then all cells are the same width based on the largest cell width
+    init(cells: [ScrollableSegmentedControlCell], configuration: ScrollableSegmentedControlConfiguration) {
+        self.cells = cells
+        self.configuration = configuration
+        super.init(frame: .zero)
+        if configuration.isProportionalWidth {
+            configureProportionalWidthCells()
+        } else {
+            configureEqualWidthCells()
+        }
+        configureHighlightLine()
+    }
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    // MARK: - Life cycle
+    override func layoutSubviews() {
```

Comment:
@eviltofu Quick comment! can you leave a single linebreak before each func and group of class/struct var/let(s)?